config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show", as: "user"
+  get "/users/:user_id/posts", to: "posts#index", as: "user_posts"
+  get "/users/:user_id/posts/:id", to: "posts#show", as: "user_post"
```

Comment:
- [ ] [Optional] Great work implementing the routes for the different controllers and their actions. I would like to suggest a recommended method for setting up routes for controllers. 

```Ruby on Rails
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

- Using this method encourages developers to use RESTFUL URL patterns when setting up routes, This concept enables you to manipulate `HTTP` verbs. 👍 
- You can read this [article](https://launchschool.com/books/demystifying_rails/read/routes_and_resources) to have a better understanding of this concept. You can also run `rails routes -c Users` or `rails routes -C Posts` to see the routes created. 👍